consumer/config.go

```diff
@@ -5,39 +5,61 @@ import (
 
 	"github.com/Shopify/sarama"
 	"github.com/heetch/felice/codec"
+	"gopkg.in/retry.v1"
 )
 
 // Config is used to configure the Consumer.
 type Config struct {
 	*sarama.Config
 
+	// KafkaAddrs holds kafka brokers addresses. There must be at least
+	// one entry in the slice.
+	// Default to localhost:9092.
+	KafkaAddrs []string
+
 	// MaxRetryInterval controls the maximum length of time that
 	// the Felice consumer will wait before trying to
 	// consume a message from Kafka that failed the first time around.
-	// The default value is 5 seconds.
+	// Default to 5 seconds.
 	MaxRetryInterval time.Duration
-
-	// Codec used to decode the message key. Defaults to codec.String.
+	// Codec used to decode the message key.
+	// Default to codec.String.
 	KeyCodec codec.Codec
+
+	retryStrategy retry.Strategy
 }
 
 // NewConfig creates a config with sane defaults.
-// The Sarama Cluster group mode will always be overwritten by the consumer
-// and thus cannot be changed, as the consumer is designed to use the ConsumerModePartitions mode.
-func NewConfig(clientID string) Config {
+// Broker addresses defaults to localhost:9092.
+func NewConfig(clientID string, addrs ...string) Config {
 	var c Config
 
 	c.Config = sarama.NewConfig()
 	c.ClientID = clientID
 	c.Consumer.Return.Errors = true
 	// Specify that we are using at least Kafka v1.0
 	c.Version = sarama.V1_0_0_0
-
 	// Distribute load across instances using round robin strategy
 	c.Consumer.Group.Rebalance.Strategy = sarama.BalanceStrategyRoundRobin
 
 	// Felice consumer configuration
+	c.KafkaAddrs = addrs
+	if c.KafkaAddrs == nil {
+		c.KafkaAddrs = []string{"localhost:9092"}
+	}
+
 	c.MaxRetryInterval = 5 * time.Second
 	c.KeyCodec = codec.String() // defaults to String
+
+	// Note: the logic in handleMsg assumes that
```

Comment:
Marvelous!  :1st_place_medal:

Reply:
All credits go to @rogpeppe for this one. 🙈 

(It's a copy paste from the consumer.go file)